pdc_client/__init__.py

```diff
@@ -89,8 +94,63 @@ def read_config_file(server_alias):
     return result
 
 
+class MultiPageBeanBag(BeanBag):
+
+    def is_multi_page_response(self, response):
+        """
+        Check if response is a dict or not
+        """
+        if isinstance(response, dict):
+            if set([u'count', u'next', u'results', u'previous']).issubset(set(response.keys())):
+                return True
+        return False
+
+    def url2path_ext(self, url):
+        """
+        @param url
+        returns path, params
+        """
+        parsed = urlparse(url)
+        return parse_qs(parsed.query)
+
+    def call(self, path, *args, **kwargs):
+        """
+        Re-write the call function of BeanBag.
+        Let it default get multi pages data if there is no page parameter.
+        """
+        try:
+            if len(args) == 0:
+                verb, body = "GET", None
+                result = self.make_request(path, verb, kwargs, body)
+
+                if self.is_multi_page_response(result) and ("page" not in kwargs.keys()):
+                    count = result["count"]
+                    multi_result = result['results']
+                    while result['next']:
+                        next_page = self.url2path_ext(result['next'])
+                        kwargs["page"] = next_page["page"][0]
+                        result = self.make_request(path, verb, kwargs, body)
+                        multi_result.extend(result['results'])
+                    if len(multi_result) != count:
+                        raise BeanBagException("MultiPageRequest failed: got %d elements expected %s" %
+                                               (len(multi_result), count), result, (path))
+                    return multi_result
+                return result
+
+            elif len(args) == 1:
+                verb, body = "POST", args[0]
+            elif len(args) == 2:
+                verb, body = args
+            else:
+                raise TypeError("__call__ expected up to 2 arguments, got %d" % (len(args)))
+            return self.make_request(path, verb, kwargs, body)
+        except Exception:
+            raise
```

Comment:
After discuss with @simozhan , I will make the change in make_request function as the original patch.